src/NotificationTargetKnowbaseItem.php

```diff
@@ -0,0 +1,206 @@
+        } else {
+            $this->data['##knowbaseitem.categories##']      = '';
+        }
+        $this->data['##knowbaseitem.is_faq##']      = Dropdown::getYesNo($knowbase->fields['is_faq']);
+        $this->data['##knowbaseitem.begin_date##']      = $knowbase->fields['begin_date'];
+        $this->data['##knowbaseitem.end_date##']      = $knowbase->fields['end_date'];
+
+        $documents = new Document_Item();
+        $associateddocuments = $documents->find([
+            'items_id' => $knowbase->getID(),
+            'itemtype' => 'KnowbaseItem'
+        ]);
+        $this->data['##knowbaseitem.numberofdocuments##']      = count($associateddocuments);
+        foreach ($associateddocuments as $docid) {
+            $document = Document::getById($docid['documents_id']);
+            $this->data['documents'][] = [
+                '##document.downloadurl##'             => $document->getDownloadLink(),
+                '##document.url##'                     => $document->getLink(),
+                '##document.filename##'                => $document->fields['filename'],
+                '##document.weblink##'                 => $document->fields['link'],
+                '##document.id##'                      => $document->getID(),
+                '##document.heading##'                 => $document->fields['name'],
+                '##document.name##'                    => $document->fields['name']
+            ];
+        }
+    }
+
+    public function getTags()
+    {
+        $tags = [
+            'knowbaseitem.url'                      => __('URL'),
+            'knowbaseitem.categories'               => __('Categories'),
+            'knowbaseitem.content'                  => __('Content'),
+            'knowbaseitem.subject'                  => __('Subject'),
+            'knowbaseitem.begin_date'               => __('Visible since'),
+            'knowbaseitem.end_date'                 => __('Visible until'),
+            'knowbaseitem.is_faq'                   => __('FAQ'),
+            'knowbaseitem.numberofdocuments'        => __('Number of documents'),
+            'document.name'                         => __('Document name'),
+            'document.downloadurl'                  => __('Document download URL'),
+            'document.url'                          => __('Document URL'),
+            'document.filename'                     => __('Document filename'),
+            'document.weblink'                      => __('Document weblink'),
+            'document.id'                           => __('Document ID'),
+            'document.heading'                      => _n('Document heading', 'Documents headings', 2),
+            'target.url'                            => __('URL'),
+            'target.name'                           => __('Name'),
+            'target.itemtype'                       => _n('Type', 'Types', 2),
+        ];
```

Comment:
as discussed yesterday, 
the ```KnowBaseItem``` ```tags``` are available for all  events ```newknowbase```, ```updateknowbase```, ```deletingknowbase```.

The others (targets* / document*) are only available for ```newknowbase```, ```updateknowbase``` events